src/AppInstallerCommonCore/Runtime.cpp

```diff
@@ -329,69 +337,86 @@ namespace AppInstaller::Runtime
         s_runtimePathStateName.emplace(std::move(suitablePathPart));
     }
 
+    void PathDetails::SetOwner(ACEPrincipal owner)
+    {
+        Owner = owner;
+        ACL[owner] = ACEPermissions::All;
+    }
+
     bool PathDetails::ShouldApplyACL() const
     {
         // Could be expanded to actually check the current owner/ACL on the path, but isn't worth it currently
-        return (CurrentUser != ACEPermissions::None || Admins != ACEPermissions::None);
+        return !ACL.empty();
     }
 
     void PathDetails::ApplyACL() const
     {
-        THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), CurrentUser == ACEPermissions::Owner && Admins == ACEPermissions::Owner);
+        bool hasCurrentUser = ACL.count(ACEPrincipal::CurrentUser) != 0;
+        bool hasSystem = ACL.count(ACEPrincipal::System) != 0;
 
-        ULONG entriesCount = 0;
-        EXPLICIT_ACCESS_W explicitAccess[2];
+        // Configuring permissions for both CurrentUser and SYSTEM while not having owner set as one of them is not valid because
+        // below we use only the owner permissions in the case of running as SYSTEM.
+        if ((hasCurrentUser && hasSystem) &&

```

Comment:
>(hasCurrentUser && hasSystem) [](http://example.com/codeflow?start=12&length=29)

If this check is only for scenarios when running as system, should we make the condition in parity with later usage

(hasCurrentUser && hasSystem && EqualSid(userToken->User.Sid, systemSID.get())

Or just throw when it's actually used.

Reply:
The point is to discover a configuration that would be illegal in that case without actually needing to be running as `SYSTEM`.